test/flag_anomaly.py

```diff
@@ -0,0 +1,126 @@
+    neighborList.update(atoms)
+    matrix = neighborlist.get_connectivity_matrix(neighborList.nl).toarray()
+    return matrix
+
+def find_max_movement(init_atoms, final_atoms):
+    '''
+    Given ase.Atoms objects, find the furthest distance that any single atom in
+    a set of atoms traveled (in Angstroms)
+    Args:
+            init_atoms      `ase.Atoms` of the structure in its initial state
+            final_atoms     `ase.Atoms` of the structure in its final state
+    Returns:
+            max_movement    A float indicating the further movement of any single atom
+                            before and after relaxation (in Angstroms)
+    '''
+    # Calculate the distances for each atom
+    distances = final_atoms.positions - init_atoms.positions
+
+    # Reduce the distances in case atoms wrapped around (the minimum image convention)
+    _, movements = find_mic(distances, final_atoms.cell, final_atoms.pbc)
+    max_movement = max(movements)
+    return max_movement
+
+def is_adsorbate_placed_correct(adslab, atoms_tag):
+    """
+    Make sure all adsorbate atoms are connected after placement.
+    False means there is at least one isolated adsorbate atom.
+    Args:
+            adslab             `ase.Atoms` of the structure in its initial state
+            atoms_tag (list)    0=bulk, 1=surface, 2=adsorbate
+    Returns:
+            boolean    If there is any stand alone adsorbate atoms after placement,
+                       return False.
+
+    """
+    adsorbate_idx = [idx for idx, tag in enumerate(self.atoms_tag) if tag==2]
+    connectivity = get_connectivity(self.init_atoms[adsorbate_idx])
+    if np.any(np.sum(connectivity, axis=0)==0):
+        return False
+    return True
+
+class DetectAnomaly:
+    def __init__(self, init_atoms, final_atoms, atoms_tag):
+        """
+        flag anomalies based on initial and final stucture of a relaxation.
+        Args:
+               init_atoms         `ase.Atoms` of the adslab in its initial state
+               final_atoms        `ase.Atoms` of the adslab in its final state
+               atoms_tag (list)    0=bulk, 1=surface, 2=adsorbate
+        """
+        self.init_atoms = init_atoms
+        self.final_atoms = final_atoms
+        self.atoms_tag = atoms_tag
+
+    def _is_adsorbate_dissociated(self):
+        """
+        True if the adsorbate is dissociated
+        """
+        # test to see if adsorbate is dissociated from surface
+        adsorbate_idx = [idx for idx, tag in enumerate(self.atoms_tag) if tag==2]
+
+        # if adsorbate is dissociated, the connectivity matrix would change
+        initial_connectivity = get_connectivity(self.init_atoms[adsorbate_idx])
+        final_connectivity = get_connectivity(self.final_atoms[adsorbate_idx])
+        if np.array_equal(initial_connectivity, final_connectivity):
+            return False
+        return True
+
+    def _is_surface_reconstructed(self, slab_movement_thres):
```

Comment:
would be good to have a default value of `slab_movement_thres`

Reply:
Done! change will be reflected in the upcoming push